docs/plugins/integrations/sync.rst

```diff
@@ -0,0 +1,77 @@
+
+The sync is initiated using the ``mautic:integrations:sync`` command. For example::
+
+    php bin/console mautic:integrations:sync HelloWorld --start-datetime="2020-01-01 00:00:00" --end-datetime="2020-01-02 00:00:00".
+
+------
+
+Registering the Integration for the Sync Engine
+===============================================
+To tell the IntegrationsBundle that this integration provides a syncing feature, tag the integration or support class with ``mautic.sync_integration`` in the plugin's ``app/config.php``.
+
+.. code-block:: php
+
+    <?php
+    return [
+        // ...
+        'services' => [
+            // ...
+            'integrations' => [
+                // ...
+                'helloworld.integration.sync' => [
+                    'class' => \MauticPlugin\HelloWorldBundle\Integration\Support\SyncSupport::class,
+                    'tags'  => [
+                        'mautic.sync_integration',
+                    ],
+                ],
+                // ...
+            ],
+            // ...
+        ],
+        // ...
+    ];
+
+
+.. compound::
+
+    The ``SyncSupport`` class must implement::
+
+        \Mautic\IntegrationsBundle\Integration\Interfaces\SyncInterface.
+
+Syncing
+=======
+
+The Mapping Manual
+__________________
+The mapping manual tells the sync engine which integration should be synced with which Mautic object (contact or company), the integration fields that were mapped to Mautic fields, and the direction the data is supposed to flow. 
+
+See https://github.com/mautic/plugin-helloworld/blob/mautic-4/Sync/Mapping/Manual/MappingManualFactory.php
+
+The Sync Data Exchange
+______________________
+This is where the sync takes place and is executed by the ``mautic:integrations:sync`` command. Mautic and the integration will build their respective reports of new or modified objects then execute the order from the other side.
+
+See https://github.com/mautic/plugin-helloworld/blob/mautic-4/Sync/DataExchange/SyncDataExchange.php
+
+Building Sync Report
+____________________
+The sync report tells the sync engine what objects are new and/or modified between the two timestamps given by the engine (or up to the integration discretion if it is a first time sync). Objects should be processed in batches which can be done using the ``RequestDAO::getSyncIteration()``. The sync engine will execute ``SyncDataExchangeInterface::getSyncReport()`` until a report comes back with no objects.
+
+If the integration supports field level change tracking, it should tell the report so that the sync engine can merge the two data sets more accurately. 
```

Comment:
```suggestion
If the Integration supports field level change tracking, it should tell the report so that the sync engine can merge the two data sets more accurately. 
```